analytics/pubxai/config/config.go

```diff
@@ -0,0 +1,92 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/prebid/prebid-server/v3/util/task"
+)
+
+type Configuration struct {
+	PublisherId        string `json:"publisher_id"`
+	BufferInterval     string `json:"buffer_interval"`
+	BufferSize         string `json:"buffer_size"`
+	SamplingPercentage int    `json:"sampling_percentage"`
+}
+
+type ConfigService interface {
+	Start(stop <-chan struct{}) <-chan *Configuration
+	IsSameAs(a *Configuration, b *Configuration) bool
+}
+
+type ConfigServiceImpl struct {
+	task       *task.TickerTask
+	configChan chan *Configuration
+}
+
+func NewConfigService(httpClient *http.Client, pubxId, endpoint, refreshInterval string) (ConfigService, error) {
+	refreshDuration, err := time.ParseDuration(refreshInterval)
+	if err != nil {
+		return nil, fmt.Errorf("fail to parse the module args, arg=analytics.pubxai.configuration_refresh_delay: %v", err)
+	}
+
+	endpointUrl, err := url.Parse(endpoint + "/config?pubxId=" + pubxId)
```

Comment:
You must sanitize the user provided pubxId string when including it as a query parameter.